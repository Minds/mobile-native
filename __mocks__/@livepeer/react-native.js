// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
