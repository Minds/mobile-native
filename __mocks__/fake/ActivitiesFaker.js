function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
