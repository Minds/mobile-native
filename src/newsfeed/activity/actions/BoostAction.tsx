import React, { PureComponent } from 'react';

import { Text, View, StyleSheet, TouchableHighlight } from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

import withPreventDoubleTap from '../../../common/components/PreventDoubleTap';
import i18n from '../../../common/services/i18n.service';
import ThemedStyles from '../../../styles/ThemedStyles';
import ActivityModel from '../../../newsfeed/ActivityModel';

// prevent double tap in touchable
const TouchableHighlightCustom = withPreventDoubleTap(TouchableHighlight);

type PropsType = {
  navigation: any;
  entity: ActivityModel;
};

/**
 * Boost Action Component
 */
export default class BoostAction extends PureComponent<PropsType> {
  /**
   * Render
   */
  render() {
    const theme = ThemedStyles.style;
    return (
      <View>
        <TouchableHighlightCustom
          style={[
            theme.rowJustifyCenter,
            theme.paddingHorizontal3x,
            theme.paddingVertical4x,
            theme.alignCenter,
          ]}
          underlayColor="transparent"
          onPress={this.openBoost}>
          <View style={[theme.rowJustifyStart, theme.centered]}>
            <Icon
              name="trending-up"
              style={[theme.fontS, theme.colorSecondaryText]}
            />
            <Text
              style={[styles.text, theme.colorSecondaryText]}
              numberOfLines={1}>
              {i18n.t('boost')}
            </Text>
          </View>
        </TouchableHighlightCustom>
      </View>
    );
  }

  /**
   * Open boost screen
   */
  openBoost = () => {
    this.props.navigation.push('Boost', { entity: this.props.entity });
  };
}

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Roboto',
    fontSize: 12,
    textAlignVertical: 'center',
    lineHeight: 21,
    letterSpacing: 1,
    marginLeft: 3,
    fontWeight: '500',
  },
});
