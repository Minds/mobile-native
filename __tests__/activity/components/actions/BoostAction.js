import 'react-native';
import React from 'react';
import { TouchableHighlight } from 'react-native';
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import BoostAction from '../../../../src/newsfeed/activity/actions/BoostAction';

// prevent double tap in touchable

describe('Boost action component', () => {
  let screen;
  beforeEach(() => {
    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = renderer.create(
      <BoostAction
        entity={activityResponse.activities[0]}
        navigation={navigation}
      />,
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    expect(screen).toMatchSnapshot();
  });

  xit('should have A boost button', async () => {
    expect(screen.root.findByType('Text').props['children']).toBe('Boost');
  });

  it('should navigate to boost on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = {
      navigate: jest.fn(),
      push: jest.fn(),
    };
    let entity = activityResponse.activities[0];
    screen = shallow(<BoostAction entity={entity} navigation={navigation} />);
    screen.update();
    let touchables = screen.find('preventDoubleTap(TouchableHighlight)');
    touchables.at(0).props().onPress();
    jest.runAllTimers();

    expect(navigation.push).toHaveBeenCalled();

    expect(screen.find('preventDoubleTap(TouchableHighlight)')).toHaveLength(1);
  });
});
