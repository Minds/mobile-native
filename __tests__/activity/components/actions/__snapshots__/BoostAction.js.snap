// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boost action component renders correctly 1`] = `
<View>
  <View
    onPress={[Function]}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    underlayColor="transparent"
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              undefined,
              undefined,
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°”µ
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Roboto",
              "fontSize": 12,
              "fontWeight": "500",
              "letterSpacing": 1,
              "lineHeight": 21,
              "marginLeft": 3,
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        Boost
      </Text>
    </View>
  </View>
</View>
`;
